diamond-api/src/render.rs

```diff
@@ -166,26 +188,62 @@ pub fn render_handler(
 
     match Interval::new(params.from, params.until).map_err(err_msg) {
         Ok(interval) => {
-            let response: Result<Vec<RenderResponceEntry>, ResponseError> = params
+            let response: Result<Vec<RenderResponseEntry>, ResponseError> = params
                 .target
                 .into_iter()
-                .map(|x| {
-                    walk(&dir, &x, interval).map(|datapoints| RenderResponceEntry {
+                .map(|metric| {
+                    // TODO Pass proper Walker via Args (or more generic Context)
+                    let w = Walker::File(state.path.clone());
+                    let f = walker(w);
+                    f(&metric, interval).map(|datapoints| RenderResponseEntry {
                         datapoints,
-                        target: x,
+                        target: metric,
                     })
                 })
                 .collect();
 
             match response {
-                Ok(response) => result(Ok(HttpResponse::Ok().json(response))),
+                Ok(response) => result(Ok(format_response(response, params.format))),
                 Err(e) => err(e.into()),
             }
         }
         Err(e) => err(e),
     }
 }
 
+// TODO: Extract to response_format_csv module
+trait IntoCsv {
```

Comment:
Other approach could be

```
pub struct Csv(pub String);

impl Into<Csv> for ... {...
```

Reply:
Done

Reply:
Why do you need generic here? One type is converted to another.

Reply:
I was thinking separation of concerns. Making `to_csv` method a part of public API of `struct RenderResponseEntry` doesn't look right.
With trait I can separate CSV-related stuff, and provide generic impl for Vec<T> when T: IntoCsv. Then one `impl IntoCsv for RenderResponseEntry` is enough to provide respective impl for Vec<RenderResponseEntry>.

Reply:
Let's keep it as it is right now. We should not block PR because of these.